src/app-layout/drawer/index.tsx

```diff
@@ -42,90 +24,143 @@ export interface DesktopDrawerProps {
+      drawerMainLabel: labels?.content,
+      drawerOpenLabel: labels?.triggerButton,
+      drawerCloseLabel: labels?.closeButton,
+    });
+    const { drawerMainLabel, drawerCloseLabel } = getDrawersLabels(drawersAriaLabels);
 
-  const regularOpenButton = (
-    <TagName ref={openButtonWrapperRef} aria-label={mainLabel} className={styles.toggle} aria-hidden={isOpen}>
-      <AppLayoutButton
-        ref={toggleRefs.toggle}
-        className={toggleClassName}
-        iconName={iconName}
-        ariaLabel={openLabel}
-        onClick={() => onToggle(true)}
-        ariaExpanded={false}
-      />
-    </TagName>
-  );
+    const regularOpenButton = (
+      <TagName ref={openButtonWrapperRef} aria-label={mainLabel} className={styles.toggle} aria-hidden={isOpen}>
+        <AppLayoutButton
+          ref={toggleRefs.toggle}
+          className={toggleClassName}
+          iconName={iconName}
+          ariaLabel={openLabel}
+          onClick={() => onToggle(true)}
+          ariaExpanded={false}
+        />
+      </TagName>
+    );
 
+    return (
+      <div
+        ref={ref}
+        className={clsx(styles.drawer, {
+          [styles['drawer-closed']]: !isOpen,
+          [testutilStyles['drawer-closed']]: !isOpen,
+          [styles['drawer-mobile']]: isMobile,
+        })}
+        style={{ width: drawerContentWidth }}
+        onBlur={
+          onLoseFocus
+            ? e => {
+                if (!e.relatedTarget || !e.currentTarget.contains(e.relatedTarget)) {
+                  onLoseFocus(e);
+                }
+              }
+            : undefined
+        }
+        onClick={event => {
+          if (onClick) {
+            onClick(event);
+          }
+          if (!isOpen) {
+            // to prevent calling onToggle from the drawer when it's called from the toggle button
+            if (
+              openButtonWrapperRef.current === event.target ||
+              !openButtonWrapperRef.current?.contains(event.target as Node)
+            ) {
+              onToggle(true);
+            }
+          }
+        }}
+      >
+        <div
+          style={{ width: drawerContentWidth, top: topOffset, bottom: bottomOffset }}
+          className={clsx(styles['drawer-content'], contentClassName)}
+        >
+          {!isMobile && regularOpenButton}
+          {resizeHandle}
+          <TagName aria-label={drawers ? drawerMainLabel : mainLabel} aria-hidden={!isOpen}>
+            <CloseButton
+              ref={toggleRefs.close}
+              className={closeClassName}
+              ariaLabel={drawers ? drawerCloseLabel : closeLabel}
+              onClick={() => {
+                onToggle(false);
+                drawers?.onChange({ activeDrawerId: undefined });
+              }}
+            />
+            {children}
+          </TagName>
+        </div>
+      </div>
+    );
+  }
+);
+
+export function DrawerTriggersBar({ isMobile, topOffset, bottomOffset, drawers, contentClassName }: DrawerTriggersBar) {
```

Comment:
`DrawerTriggersBar` is used twice as a function name and as props type. I am very surprised it did not cause any problems because the function is exported but the type is not. 

To avoid all such problems we always call prop types with `Props` ending,  `DrawerTriggersBarProps`